src/cpp/rtps/history/CacheChangePool.cpp

```diff
@@ -118,8 +120,7 @@ bool CacheChangePool::reserve_Cache(CacheChange_t** chan, uint32_t dataSize)
                     return false;
                 }
             }
-            *chan = m_freeCaches.back();
-            m_freeCaches.erase(m_freeCaches.end()-1);
+            *chan = m_freeCaches.pop_back();
```

Comment:
```suggestion
            *chan = m_freeCaches.back();
            m_freeCaches.pop_back();
```
That should be the way

Reply:
done. AND checked it compiles :S